dozer/cogs/filter.py

```diff
@@ -0,0 +1,259 @@
+import discord
+import re
+from discord.ext.commands import guild_only, has_permissions
+import datetime
+
+from ._utils import *
+from .. import db
+
+
+class Filter(Cog):
+    """The filters need to be compiled each time they're run, but we don't want to compile every filter
+    Every time it's run, or all of them at once when the bot starts. So the first time that filter is run,
+    the compiled object is placed in here. This dict is actually a dict full of dicts, with each parent dict's key
+    being the guild ID for easy accessing.
+    """
+    filter_dict = {}
+
+    """Helper Functions"""
+
+    async def check_dm_filter(self, ctx, embed):
+        with db.Session() as session:
+            results = session.query(WordFilterSetting).filter_by(guild_id=ctx.guild.id, setting_type="dm") \
+                .one_or_none()
+
+            if results is None:
+                results = True
+            else:
+                results = results.value
+
+            if results:
```

Comment:
`"0"` is truey, so this condition will pass no matter what is read from the database.